create-hpc-cluster-linux-cn/azuredeploy.json

```diff
@@ -0,0 +1,455 @@
+        "dnsName": "[concat(parameters('clusterName'), 'ep')]",
+        "nicName": "[concat(parameters('clusterName'),'-nic')]",
+        "nicNameHN": "[concat(variables('nicName'),'-hn')]",
+        "linuxImagePublishers": {
+            "CentOS_6.6": "OpenLogic",
+            "CentOS_7.0": "OpenLogic",
+            "SLES_12": "SUSE",
+            "SLES_Premium_12": "SUSE",
+            "SLES_HPC_12": "SUSE",
+            "SLES_HPC_Premium_12": "SUSE",
+            "Ubuntu_14.04": "Canonical"
+        },
+        "linuxImageOffers": {
+            "CentOS_6.6": "CentOS",
+            "CentOS_7.0": "CentOS",
+            "SLES_12": "SLES",
+            "SLES_Premium_12": "SLES-Priority",
+            "SLES_HPC_12": "SLES-HPC",
+            "SLES_HPC_Premium_12": "SLES-HPC-Priority",
+            "Ubuntu_14.04": "UbuntuServer"
+        },
+        "linuxImageSkus": {
+            "CentOS_6.6": "6.6",
+            "CentOS_7.0": "7.0",
+            "SLES_12": "12",
+            "SLES_Premium_12": "12",
+            "SLES_HPC_12": "12",
+            "SLES_HPC_Premium_12": "12",
+            "Ubuntu_14.04": "14.04.3-LTS"
+        },
+        "computeNodeImagePublisher": "[variables('linuxImagePublishers')[parameters('computeNodeImage')]]",
+        "computeNodeImageOffer": "[variables('linuxImageOffers')[parameters('computeNodeImage')]]",
+        "computeNodeImageSku": "[variables('linuxImageSkus')[parameters('computeNodeImage')]]",
+        "adminBase64Password": "[base64(parameters('adminPassword'))]",
+        "customScriptLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/create-hpc-cluster/",
+        "iaasInfoArg": "[concat('-SubscriptionId ', subscription().subscriptionId, ' -VNet ', variables('virtualNetworkName'), ' -Subnet ', variables('subnet1Name'), ' -Location \"', resourceGroup().location, '\"')]",
+        "headNodeCommandStr": "[concat('powershell.exe -ExecutionPolicy Unrestricted -File PrepareHN.ps1 -DomainFQDN ', parameters('privateDomainName'), ' -AdminUserName ', parameters('adminUsername'), ' -AdminBase64Password \"', variables('adminBase64Password'), '\" -PostConfigScript \"', parameters('headNodePostConfigScript'), '\" -UnsecureDNSUpdate ', variables('iaasInfoArg'))]",
+        "storageConnStrPrefix": "[concat('DefaultEndpointsProtocol=https;AccountName=', variables('storageAccountName'), ';AccountKey=')]",
+        "computeNodeCommandStr": "[concat('powershell.exe -ExecutionPolicy Unrestricted -File PrepareCN.ps1 -DomainFQDN ', parameters('privateDomainName'), ' -ClusterName ', parameters('clusterName'),' -AdminUserName ', parameters('adminUsername'), ' -AdminBase64Password \"', variables('adminBase64Password'), '\"')]"
+    },
+
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[variables('storageAccountName')]",
+            "apiVersion": "2015-05-01-preview",
```

Comment:
Make apiVersion a variable so that it becomes easy to update this template for newer API versions.